tools/hardware/memory.py

```diff
@@ -17,10 +17,6 @@ def __init__(self, base: int, size: int, owner: 'WrappedNode' = None):
         self.size = size
         self.owner = owner
 
-    @staticmethod
-    def clone(other):
-        return Region(other.base, other.size)
-
```

Comment:
Why is it safe to remove `clone()`?

Reply:
Ok, found the commit message, sorry. How sure are we that this is only used in this repo? (I assume that is the scope used to find out that it wasn't called).

I agree that the clone functionality looks a bit strange -- `clone` potentially was added initially and not updated when `owner` was added, which would also need to support `clone` properly.